### ISSUE TITLE
Compile error with TMVCActiveRecord and Delphi 10.3

### ISSUE THREAD

**Issue by gmurt** [danieleteti/delphimvcframework#763]
I'm trying to integrate DMVC, and specifically TMVCActiveRecord into an existing project but as soon as I create an entity class which inherits from TMVCActiveRecord, I get the following internal error when compiling...

[dcc32 Fatal Error] Data.Entities.pas(24): F2084 Internal Error: URW1237

I'm on an active subscription and Develop new projects in Delphi 12 but this is a legacy project and upgrading to D12 is not an option due to the number of third party products used.

Has any one else experienced this issue?

Here is my code, if I comment out the class then the project compiles fine. 

````
unit Data.Entities;

interface

uses
  Classes,
  MVCFramework.ActiveRecord,
  MVCFramework.Nullables;


type
  TTestEntity = class(TMVCActiveRecord)
  public

  end;

implementation

uses
  SysUtils;

end.
````

**Comment by danieleteti** [comment]
That's is a Delphi problem... however....
Did you try to comment piece by piece? For instance, commenting the nullables unit, change something?
Such kind of internal errors are raised for some internal compiler bug, so is quite difficult to understand what causes them. Try to comment something to understand what is causing it. Older DMVCFramework versions raise the same error? Could you use an older version? Can this test be useful to understand what is the main error cause?

**Comment by gmurt** [comment]
Hi Daniel,

Yes, it's the following line which causes the issue. Even in a blank Delphi project, the error occurs.

TTestEntity = class(TMVCActiveRecord) 

I did some googling and it's something to do with generics and overloading so I guessed it's probably not an easy fix.

I spent the evening modifying my code (and upgrading some third party commercial libraries) and now I have my project compiling in 11.3 - this didn't turn out to be as big a challenge as I expected.

Kind regards,
Graham
